flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/utils/SourceCodeGenerator.java

```diff
@@ -381,10 +356,10 @@ private String generateResourceFunction(Function function) {
 
         String template = "%s\tresource function %s %s(%s) returns %s {" +
                 "%n\t\tdo {" +
-                "%n\t\t\treturn self.%s;" +
+                "%n\t\t\tpanic error(\"Unimplemented function\");" +
                 "%n\t\t} on fail error err {" +
```

Comment:
This generates an unused variable diagnostic, right? Can't we just use `on fail`?

Reply:
No need to address with this PR

Reply:
That's okay since it's a warning.